### PR TITLE
Fix superflous shader setting updates

### PR DESCRIPTION
**Pull request by gregorycu**
Improve rendering performance by ~40%

### CODE REVIEW

**Comment by nerzhul** [comment]
seems better but travis failed. Please look at travis logs and fix it

**Comment by gregorycu** [comment]
Will do.

File: src/sky.h [hunk 1]
```diff
@@ -137,5 +137,19 @@ class Sky : public scene::ISceneNode
 	video::ITexture* m_moon_tonemap;
 };
 
+class SkyBackgroundColorProvider
```
Comment (kahrl) [inline src/sky.h]: What's the benefit of this class over a Sky* variable?

**Comment by Zeno-** [comment]
Does not build. Please make a new PR when it does.

**Comment by nerzhul** [comment]
#2375 override this

**Pull request closed without merging**